File: 1548-check-if-all-1s-are-at-least-length-k-places-away/check-if-all-1s-are-at-least-length-k-places-away.js
var kLengthApart = function(nums, k) {
    let last1 = -1
    for(let i = 0; i <nums.length; i++){
        if(nums[i] == 1){
        last1 = i;
        break;
        }
    }
    if (last1 == -1)
    return true;
    
    let start = last1 + 1;
    for(let i = start;i<nums.length;i++){
            if (nums[i] == 1){
                if (((i - last1) - 1) >= k){
                    last1 = i;
                    continue;
                }
                else{
                    return false;
                }
            }
    }
    return true;
};

File: 1548-check-if-all-1s-are-at-least-length-k-places-away/check-if-all-1s-are-at-least-length-k-places-away.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        last1Index = -1
        for i in range(0,len(nums)):
            if nums[i] == 1:
                last1Index = i
                break
        if last1Index == -1:
            return True
        start = last1Index + 1
        for i in range(start,len(nums)):
            if nums[i] == 1:
                if ((i - last1Index) - 1) >= k:
                    last1Index = i
                    continue 
                else:
                    return False
        return True